### PR TITLE
fix: responsive tables for reaction tables on mobile devices

### LINKED ISSUE #1199: Reactions table not useful on mobile devices
**Describe the bug**

Some reaction tables are hard to read / overview on a mobile phone, the user will just scroll through what looks like an empty table.

**Steps to reproduce**

- Visit eg `/explore/Human-GEM/interaction-partners/ENSG00000122687` from a mobile device / with a narrow window.

**Expected behavior**

- [ ] Tables look good on mobile devices

**Acceptance criteria**
- [ ] Check other tables as well in the Gem Browser.
 

**Screenshots**

**Additional context**

Created from #1154.
**Note**: the original example for this (ENSG00000145949) is not useful anymore as the data models have been updated since. It created a much, much longer section of nothingness on the screen that the now provided example.

### PR DESCRIPTION
**Pull request by MalinAhlberg**
**Related issue(s) and PR(s)**  
This PR closes #1199 .

**Type of change**  

- [x] Bug fix (non-breaking change which fixes an issue)

**List of changes made**  

On narrow screens and mobile devices, the styling of the reaction tables is changed so that all fields can be seen and so that the screen is not filled with white space.
Thanks to @e0 for helping out with css!

**Screenshot of the fix**  

2022-12-05-154948_354x578_scrot

**Testing**  

- Please visit different pages from Interaction Partners, using different screen sizes, and make sure that the reaction table looks good.

**Further comments**  

This issue also includes checking the **reaction tables in the GEM Browser**. They do already have similar styling for narrow screens, see below
2022-12-05-160537_320x634_scrot

One could extend this PR to include the column header to each field for these tables, as for the Interaction Partners version in this PR. A non-finished example is shown below
2022-12-05-160446_312x414_scrot

This would require some refactoring, so a new issue is created for this (#1251).

**Definition of Done checklist**  
- [x] My code meets the Acceptance Criteria
- [x] My code follows the [NBIS style guidelines](https://github.com/NBISweden/development-guidelines)
- [x] I have performed a self-review of my own code and commented any hard-to-understand areas
- [x] Tests and lint/format validations are passing
- [x]  My changes generate no new warnings

### CODE REVIEW

**Comment by inghylt** [comment]
Nice! I think we should improve the GEM browser reaction table in a similar way. If that's seems like a somewhat complicated task, I think it's better to do this in a new issue.

However, when visiting GEM browser on this branch, the layout is not very nice: 

`/explore/Human-GEM/gem-browser/metabolite/MAM02039g`

image
image

When switching to `main` the layout is as expected. Can somebody else verify this? It's strange that there should be changes introduced to GEM browser since the related files has not been modified in this PR

File: frontend/src/components/explorer/interactionPartners/ReactionTable.vue [hunk 1]
```diff
@@ -270,4 +279,68 @@ export default {
     }
   }
 }
+table td {
+  text-align: inherit;
+}
+
+@media screen and (max-width: 600px) {
```
Comment (inghylt) [inline frontend/src/components/explorer/interactionPartners/ReactionTable.vue:286]: in `/frontend/src/style/vars.scss` we define mobile as < 660px, why was 600 chosen here?

Comment (MalinAhlberg) [reply frontend/src/components/explorer/interactionPartners/ReactionTable.vue]: Very good point! We should use ` @media screen and (max-width: $tablet)` here. Implemented in 57ee653

**Comment by MalinAhlberg** [comment]
> However, when visiting GEM browser on this branch, the layout is not very nice:
> 
> `/explore/Human-GEM/gem-browser/metabolite/MAM02039g`

That is strange! I can confirm the behavior, and will try to debug it.

**Comment by MalinAhlberg** [comment]
> > However, when visiting GEM browser on this branch, the layout is not very nice:
> > `/explore/Human-GEM/gem-browser/metabolite/MAM02039g`
> 
> That is strange! I can confirm the behavior, and will try to debug it.

This should be fixed by 79b8e0a (please verify).

**Comment by e0** [comment]
For styling, I have these recommendations:

1. Use scoped (`<style lang="scss" scoped>`). We did some [research](https://github.com/MetabolicAtlas/MetabolicAtlas/issues/917#issuecomment-1225549783) on this before and concluded that this would not decrease performance. Making this scoped would prevent the style from affecting other instances of `.reaction-table`.
2. Use [nested `scss`](https://sass-lang.com/guide#topic-3). This avoids repetition of prefixing `.reaction-table` for example.
3. Use classes or ids instead of html element tags as the starting selector. This is also mentioned in the link from the first point. I think currently there is only one instance of `table td {` that needs to be changed in this file.

**Comment by inghylt** [comment]
@MalinAhlberg I can confirm that 79b8e0a fixed the problem mentioned in https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1245#issuecomment-1339043868

**Comment by MalinAhlberg** [comment]
> For styling, I have these recommendations:
> 
> 1. Use scoped (`<style lang="scss" scoped>`). We did some [research](https://github.com/MetabolicAtlas/MetabolicAtlas/issues/917#issuecomment-1225549783) on this before and concluded that this would not decrease performance. Making this scoped would prevent the style from affecting other instances of `.reaction-table`.
> 2. Use [nested `scss`](https://sass-lang.com/guide#topic-3). This avoids repetition of prefixing `.reaction-table` for example.
> 3. Use classes or ids instead of html element tags as the starting selector. This is also mentioned in the link from the first point. I think currently there is only one instance of `table td {` that needs to be changed in this file.

Thanks for the instructions, @e0, very useful!
I tried to implemented them in d06a8c6. Please let me know if I misinterpreted anything.

**Comment by e0** [comment]
Sorry @MalinAhlberg, I should have been more specific with the descriptions. I have added [a commit](https://github.com/MetabolicAtlas/MetabolicAtlas/commit/c7510cb34cf60658647b78cec758aeea16bdc421) in a separate branch that you can take a look at. I haven't tested too carefully now so please have a double check in case I messed anything up.

When the styling is nested like this, the nesting gets transformed when it is compiled to CSS. For example the following

```scss
  .reaction-table table {
    thead {
      border: none;
    }
  }
```

becomes 

```css
  .reaction-table table thead {
    border: none;
  }
```

For the 3rd point about classes or ids vs. html element tags, I think that only applies for the first part of the selector. In this example since the first part is `.reaction-table`, it is ok to use `table` or `thead` afterwards without the performance penalty you would get from using only `table thead {`.

Please let me know if there is anything you want me to elaborate on. If you are interested you can also use this site to see how scss gets transformed to CSS: https://www.sassmeister.com

**Review by e0 — APPROVED** [review]
Looks good @MalinAhlberg!

Personally I would prefer to not change the border style if possible. Below are the changes in code and visuals. Feel free to disregard it or change as you like. Or if you want me to just push up these changes please let me know.

image

image

**Comment by MalinAhlberg** [comment]
> Looks good @MalinAhlberg!
> 
> Personally I would prefer to not change the border style if possible. Below are the changes in code and visuals. Feel free to disregard it or change as you like. Or if you want me to just push up these changes please let me know.
> 

Thanks @e0 ! I like your  version, it's in f5fb2c3.

**Comment by MalinAhlberg** [comment]
Added e6d0e64 after discussions on Slack.

**Pull request merged**